src/main/java/org/mockito/internal/invocation/InvocationsFinder.java

```diff
@@ -116,17 +120,20 @@ public static Location getLastLocation(List<Invocation> invocations) {
     }
 
     public static Invocation findPreviousVerifiedInOrder(List<Invocation> invocations, InOrderContext context) {
-        LinkedList<Invocation> verifiedOnly = ListUtil.filter(invocations, new RemoveUnverifiedInOrder(context));
+        List<Invocation> verifiedOnly = invocations
+            .stream()
+            .filter(context::isVerified)
+            .collect(Collectors.toList());
 
         if (verifiedOnly.isEmpty()) {
             return null;
         } else {
-            return verifiedOnly.getLast();
+            return verifiedOnly.get(verifiedOnly.size() - 1);
```

Comment:
This can be `.reduce((first, second) -> second).orElse(null)`

Reply:
Isnt the existing:
- more readable
- performant, _getLast_ is O(1) for array and linked list
Sorry for the silly question.